--- libraries/GCS_MAVLink/GCS_Param.cpp
+++ libraries/GCS_MAVLink/GCS_Param.cpp
@@ -264,7 +264,9 @@
     ftr = &packet.param_value; // AIS
 
     if (abrt == 1) { // AIS
-        hal.console->printf("AIS UAF: Param name not listed: %s, %f\n", *ptr, *ftr); // AIS
+        if (this->_queued_parameter == nullptr) {
+return;}
+hal.console->printf("AIS UAF: Param name not listed: %s, %f\n", *ptr, *ftr); // AIS
         return; // AIS
     } // AIS
 
