--- libraries/GCS_MAVLink/GCS_Param.cpp
+++ libraries/GCS_MAVLink/GCS_Param.cpp
@@ -252,7 +252,9 @@
     memcpy(ptr, packet.param_id, sizeof(ptr)); // AIS
 
     // find existing param so we can get the old value
-    vp = AP_Param::find(key, &var_type);
+    if (this->_queued_parameter == nullptr) {
+return;}
+vp = AP_Param::find(key, &var_type);
     if (vp == nullptr) {
         abrt = 1; // AIS
         hal.console->printf("AIS Debug: ptr freed\n"); // AIS
