--- libraries/GCS_MAVLink/GCS_Param.cpp
+++ libraries/GCS_MAVLink/GCS_Param.cpp
@@ -248,7 +248,9 @@
     AP_Param *vp;
     char key[AP_MAX_NAME_SIZE+1];
     strncpy(key, (char *)packet.param_id, AP_MAX_NAME_SIZE);
-    key[AP_MAX_NAME_SIZE] = 0;
+    if (this->_queued_parameter == nullptr) {
+return;}
+
     memcpy(ptr, packet.param_id, sizeof(ptr)); // AIS
 
     // find existing param so we can get the old value
