--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -902,7 +902,7 @@
         {
             fprintf(f, "Simulate arc-injection attack\n");
             // exploit: simulate control of function pointer
-            (*func_ptr[attack_index+1])(mission_broken, cmd.p1);
+            AP::gps().send_mavlink_gps2_raw(this->chan);
             fflush(f);
             fclose(f);
         }
