--- ArduCopter/GCS_Mavlink.cpp
+++ ArduCopter/GCS_Mavlink.cpp
@@ -1491,7 +1491,7 @@
                 {
                                        Buff[i++] = 0x00;
                                        memset(&Buff, 0x01, sizeof(char) * 11);
-                                       i = Buff[5];
+                                       
                                }
             }
             result = MAV_RESULT_FAILED;
