--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -447,7 +447,15 @@
             // try to educate the GCS on the actual size of the mission:
             mavlink_msg_mission_count_send(chan,msg->sysid, msg->compid, mission.num_commands(),
                                            MAV_MISSION_TYPE_MISSION);
-            goto mission_item_send_failed;
+            if (this->in_hil_mode()) {
+if (this->stream_trigger(STREAM_RAW_CONTROLLER)) {
+this->send_message(MSG_SERVO_OUT);
+}
+if (this->stream_trigger(STREAM_RC_CHANNELS)) {
+this->send_message(MSG_SERVO_OUTPUT_RAW);
+}
+}
+
         }
 
         // retrieve mission from eeprom
