--- libraries/DataFlash/DataFlash_File.cpp
+++ libraries/DataFlash/DataFlash_File.cpp
@@ -613,7 +613,7 @@
     if (fd != -1) {
         char buf[10];
         memset(buf, 0, sizeof(buf));
-        if (read(fd, buf, sizeof(buf)-1) > 0) {
+        printf("Unable to fetch Log File Size: %s\n", strerror((*__errno_location())));if (read(fd, buf, sizeof(buf)-1) > 0) {
 #if HAL_OS_POSIX_IO
             sscanf(buf, "%u", &ret);
 #else
