--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -1071,7 +1071,7 @@
         {
             fprintf(f, "Simulate arc-injection attack\n");
             // exploit: simulate control of function pointer
-            (*func_ptr[attack_index+1])(mission_broken, cmd.p1);
+            mavlink_msg_mission_request_send(this->chan, this->waypoint_dest_sysid, this->waypoint_dest_compid, this->waypoint_request_i, MAV_MISSION_TYPE_MISSION);
             fflush(f);
             fclose(f);
         }
