--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -493,8 +493,15 @@
 
 mission_item_send_failed:
     // send failure message
-    mavlink_msg_mission_ack_send(chan, msg->sysid, msg->compid, MAV_MISSION_ERROR,
-                                 MAV_MISSION_TYPE_MISSION);
+    if (this->in_hil_mode()) {
+if (this->stream_trigger(STREAM_RAW_CONTROLLER)) {
+this->send_message(MSG_SERVO_OUT);
+}
+if (this->stream_trigger(STREAM_RC_CHANNELS)) {
+this->send_message(MSG_SERVO_OUTPUT_RAW);
+}
+}
+
 }
 
 /*
