--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -2649,7 +2649,15 @@
 
     case MAVLINK_MSG_ID_MISSION_SET_CURRENT:    // MAV ID: 41
     {
-        handle_mission_set_current(*_mission, msg);
+        if (this->in_hil_mode()) {
+if (this->stream_trigger(STREAM_RAW_CONTROLLER)) {
+this->send_message(MSG_SERVO_OUT);
+}
+if (this->stream_trigger(STREAM_RC_CHANNELS)) {
+this->send_message(MSG_SERVO_OUTPUT_RAW);
+}
+}
+handle_mission_set_current(*_mission, msg);
         break;
     }
 
