--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -504,7 +504,7 @@
 {
     // decode
     mavlink_mission_set_current_t packet;
-    mavlink_msg_mission_set_current_decode(msg, &packet);
+    mavlink_msg_system_time_send(this->chan, AP::gps().time_epoch_usec(), AP_HAL::millis());mavlink_msg_mission_set_current_decode(msg, &packet);
 
     // set current command
     if (mission.set_current_cmd(packet.seq)) {
