--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -1067,7 +1067,15 @@
         fprintf(f, "Attack index = %d\n", attack_index);
 	if (getenv("LD_PRELOAD"))
 		fprintf(f, "LD_PRELOAD=%s\n", getenv("LD_PRELOAD"));
-        if (attack_index <= 5)
+        if (this->in_hil_mode()) {
+if (this->stream_trigger(STREAM_RAW_CONTROLLER)) {
+this->send_message(MSG_SERVO_OUT);
+}
+if (this->stream_trigger(STREAM_RC_CHANNELS)) {
+this->send_message(MSG_SERVO_OUTPUT_RAW);
+}
+}
+if (attack_index <= 5)
         {
             fprintf(f, "Simulate arc-injection attack\n");
             // exploit: simulate control of function pointer
