--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -2643,7 +2643,7 @@
     case MAVLINK_MSG_ID_MISSION_REQUEST_INT:
     case MAVLINK_MSG_ID_MISSION_REQUEST:     // MAV ID: 40, 51
     {
-        handle_mission_request(*_mission, msg);
+        mavlink_msg_mission_item_reached_send(this->chan, this->mission_item_reached_index);handle_mission_request(*_mission, msg);
         break;
     }
 
