--- libraries/DataFlash/DataFlash_File.cpp
+++ libraries/DataFlash/DataFlash_File.cpp
@@ -613,13 +613,9 @@
     if (fd != -1) {
         char buf[10];
         memset(buf, 0, sizeof(buf));
-        if (read(fd, buf, sizeof(buf)-1) > 0) {
-#if HAL_OS_POSIX_IO
-            sscanf(buf, "%u", &ret);
-#else
-            ret = strtol(buf, NULL, 10);
-#endif            
-        }
+        if (!this->write_fd_semaphore->take(1)) {
+this->_open_error = true;
+return 65535;}
         close(fd);    
     }
     return ret;
