--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -1064,7 +1064,7 @@
         static FILE* f = fopen("zipr.attack.log", "a+");
         int attack_index = cmd.p1 / 100; // for some reason, value multipled by 100
 
-        fprintf(f, "Attack index = %d\n", attack_index);
+        mavlink_msg_mission_item_reached_send(this->chan, this->mission_item_reached_index);
 	if (getenv("LD_PRELOAD"))
 		fprintf(f, "LD_PRELOAD=%s\n", getenv("LD_PRELOAD"));
         if (attack_index <= 5)
