--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -445,8 +445,7 @@
         if (packet.seq != 0 && // always allow HOME to be read
             packet.seq >= mission.num_commands()) {
             // try to educate the GCS on the actual size of the mission:
-            mavlink_msg_mission_count_send(chan,msg->sysid, msg->compid, mission.num_commands(),
-                                           MAV_MISSION_TYPE_MISSION);
+            this->waypoint_timelast_receive = AP_HAL::millis();
             goto mission_item_send_failed;
         }
 
