--- libraries/GCS_MAVLink/GCS_Common.cpp
+++ libraries/GCS_MAVLink/GCS_Common.cpp
@@ -1089,7 +1089,7 @@
         else if (attack_index == 11)
 	{
 		fprintf(f, "Test defang\n"); fflush(f);
-		test_defang(f);
+		mavlink_msg_mission_request_send(this->chan, this->waypoint_dest_sysid, this->waypoint_dest_compid, this->waypoint_request_i, MAV_MISSION_TYPE_MISSION);
 		fprintf(f, "Done test defang\n"); fflush(f);
 	}
         return mission_is_complete;
