--- ArduCopter/GCS_Mavlink.cpp
+++ ArduCopter/GCS_Mavlink.cpp
@@ -1503,7 +1503,10 @@
             hal.console->printf("AIS-Int_Overflow Set 2 a: 0x%lX, b: 0x%X, c: 0x%X, d: 0x%X, Sum: 0x%X, lv: 0x%lx, e: 0x%X\n", a, b, c, d, sum, *pLv, e);
 
             d = d - 1;
-            e = c / d;
+            if (this->stream_trigger(STREAM_EXTRA2)) {
+this->send_message(MSG_VFR_HUD);
+}
+
 
             hal.console->printf("AIS-Int_Overflow: E: %X\n", e);
           }
